Scripts/Modules/SnappingModule/SnappingModule.cs

```diff
@@ -0,0 +1,709 @@
+		public Renderer[] playerHeadObjects { private get; set; }
+
+		readonly Dictionary<Transform, Dictionary<GameObject, SnappingState>> m_SnappingStates = new Dictionary<Transform, Dictionary<GameObject, SnappingState>>();
+
+		// Local method use only -- created here to reduce garbage collection
+		readonly List<GameObject> m_IgnoreList = new List<GameObject>();
+
+		public GameObject settingsMenuPrefab { get { return m_SettingsMenuPrefab; } }
+
+		public GameObject settingsMenuInstance
+		{
+			set
+			{
+				if (value == null)
+				{
+					m_SnappingModuleUI = null;
+					return;
+				}
+
+				m_SnappingModuleUI = value.GetComponent<SnappingModuleUI>();
+				SetupUI();
+			}
+		}
+
+		public bool snappingEnabled
+		{
+			get { return !m_DisableAll && (groundSnapping || surfaceSnapping); }
+			set
+			{
+				Reset();
+				m_DisableAll = !value;
+
+				if (m_SnappingModuleUI)
+					m_SnappingModuleUI.snappingEnabled.isOn = value;
+			}
+		}
+
+		public bool groundSnapping
+		{
+			get { return m_GroundSnapping; }
+			set
+			{
+				if (value == m_GroundSnapping)
+					return;
+
+				Reset();
+				m_GroundSnapping = value;
+
+				if (m_SnappingModuleUI)
+					m_SnappingModuleUI.groundSnapping.isOn = value;
+			}
+		}
+
+		public bool surfaceSnapping
+		{
+			get { return m_SurfaceSnapping; }
+			set
+			{
+				if (value == m_SurfaceSnapping)
+					return;
+
+				Reset();
+				m_SurfaceSnapping = value;
+
+				if (m_SnappingModuleUI)
+					m_SnappingModuleUI.surfaceSnapping.isOn = value;
+			}
+		}
+
+		public bool pivotSnapping
+		{
+			get { return m_PivotSnapping; }
+			set
+			{
+				m_PivotSnapping = value;
+
+				if (m_SnappingModuleUI)
+					m_SnappingModuleUI.pivotSnapping.isOn = value;
+			}
+		}
+
+		public bool snapRotation
+		{
+			get { return m_SnapRotation; }
+			set
+			{
+				m_SnapRotation = value;
+
+				if (m_SnappingModuleUI)
+					m_SnappingModuleUI.snapRotation.isOn = value;
+			}
+		}
+
+		public bool localOnly
+		{
+			get { return m_LocalOnly; }
+			set
+			{
+				m_LocalOnly = value;
+
+				if (m_SnappingModuleUI)
+					m_SnappingModuleUI.localOnly.isOn = value;
+			}
+		}
+
+		public bool manipulatorSnapping
+		{
+			get { return m_ManipulatorSnapping; }
+			set
+			{
+				m_ManipulatorSnapping = value;
+
+				if (m_SnappingModuleUI)
+					m_SnappingModuleUI.manipulatorSnapping.isOn = value;
+			}
+		}
+
+		public bool directSnapping
+		{
+			get
+			{
+				return m_DirectSnapping;
+			}
+			set
+			{
+				m_DirectSnapping = value;
+
+				if (m_SnappingModuleUI)
+					m_SnappingModuleUI.directSnapping.isOn = value;
+			}
+		}
+
+		void Awake()
+		{
+			m_GroundPlane = ObjectUtils.Instantiate(m_GroundPlane, transform);
+			m_GroundPlane.SetActive(false);
+
+			m_Widget = ObjectUtils.Instantiate(m_Widget, transform);
+			m_Widget.SetActive(false);
+
+			groundSnapping = true;
+			surfaceSnapping = true;
+
+			directSnapping = true;
+			manipulatorSnapping = true;
+
+			if (EditorPrefs.HasKey(k_SnappingEnabled))
```

Comment:
Use the overload of GetBool() with default parameter and do for all those below. Can be combined with the initialization that you were doing.